repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,73 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledDesktopsFacts, InstalledKdeAppsFacts
+
+
+def check_kde_gnome():
+    desktopFacts = next(api.consume(InstalledDesktopsFacts))
+    kde_desktop_installed = desktopFacts.kde_installed
+    gnome_desktop_installed = desktopFacts.gnome_installed
+
+    # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+    if not kde_desktop_installed and not gnome_desktop_installed:
+        api.current_logger().info("No desktop installed. Continuing with the upgrade.")
+        return
+
+    if kde_desktop_installed:
+        api.current_logger().info("KDE desktop is installed. Checking what we can do about it.")
+        if not gnome_desktop_installed:
+            api.current_logger().error("Cannot perform the upgrade because there is"
+                                       " no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("The KDE desktop environment is not available on RHEL 8. "
```

Comment:
```suggestion
                reporting.Summary("Because the KDE desktop environment is not available on RHEL 8, all the KDE-related packages would be removed during the upgrade. There would be no desktop environment installed after the upgrade."
```

Reply:
@bocekm 
The KDE desktop environment is unvailable on RHEL 8, therefore all KDE-related packages need to be removed during the upgrade. As a result, no desktop environment would be installed after the upgrade.

I miss the GNOME part here. However, I don't have a good picture of when and among what other information the messages above are shown. So adjust as needed so that similar information is not duplicated in one place.

Reply:
Gnome is mentioned in the remediation a few lines below. The remediation message I suggest is: "Remove KDE (at least the `kde-workspace` package) or install the GNOME desktop environment to be able to upgrade."

Reply:
I've seen that and the text looks good to me. So for as long as the customer always gets the remediation hint, too, I'm fine with the wording.